src/main/java/tracko/storage/JsonAdaptedItem.java

```diff
@@ -0,0 +1,80 @@
+package tracko.storage;
+
+import java.util.HashSet;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import tracko.commons.exceptions.IllegalValueException;
+import tracko.model.items.Description;
+import tracko.model.items.Item;
+import tracko.model.items.ItemName;
+import tracko.model.items.Quantity;
+
+/**
+ * Jackson-friendly version of {@link Item}.
+ */
+public class JsonAdaptedItem {
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Item's %s field is missing!";
+
+    private final String itemName;
+    private final Integer quantity;
+    private final String description;
+
+    /**
+     * Constructs a {@code JsonAdaptedItem} with the given item details.
+     */
+    @JsonCreator
+    public JsonAdaptedItem(@JsonProperty("itemName") String itemName, @JsonProperty("quantity") Integer quantity,
+                            @JsonProperty("description") String description) {
+        this.itemName = itemName;
+        this.quantity = quantity;
+        this.description = description;
+    }
+
+    /**
+     * Converts a given {@code Item} into this class for Jackson use.
+     */
+    public JsonAdaptedItem(Item source) {
+        itemName = source.getItemName().itemName;
+        quantity = source.getQuantity().getQuantity();
```

Comment:
This sounds nitpicky, but I think `source.getQuantity().quantity` would also work because the `quantity` field is public in the `Item` class. Just to make it consistent with the others, because they use `source.getItemName().itemName` and `source.getDescription().value`.

Reply:
I tried hahahah it doesn't work 'cos the class for `quantity` is `Integer` but the class for `source.getQuantity().quantity` is `int`. I used `Integer` instead of `int` here 'cos it's easier to convert to string and back